lstchain/image/muon/muon_analysis.py

```diff
@@ -181,30 +176,31 @@ def analyze_muon_event(event_id, image, geom, equivalent_focal_length,
     candidate_clean_ring = all(
         [radial_distribution['standard_dev'] < max_radial_stdev,
          radial_distribution['excess_kurtosis'] < max_radial_excess_kurtosis,
-         npix_above_threshold(pix_ring, tailcuts[0]) > min_pix_fraction_after_cleaning * min_pix,
-         npix_composing_ring(pix_ring) > min_pix,
+         (pix_ring > tailcuts[0]).sum() >
+         min_pix_fraction_after_cleaning * min_pix,
+         np.count_nonzero(pix_ring) > min_pix,
          muonringparam.ring_radius < max_ring_radius,
          muonringparam.ring_radius > min_ring_radius
         ])
 
     if  candidate_clean_ring:
-        ctel = MuonLineIntegrate(
-            mirror_radius, hole_radius = 0.308 * u.m,
-            pixel_width=0.1 * u.deg,
-            sct_flag=False,
-            secondary_radius = 0. * u.m
-        )
+        intensity_fitter = MuonIntensityFitter()
 
-        muonintensityoutput = ctel.fit_muon(
-            muonringparam.ring_center_x,
-            muonringparam.ring_center_y,
-            muonringparam.ring_radius,
-            x[dist_mask], y[dist_mask],
-            image[dist_mask])
+        # Use same hard-coded value for pedestal fluctuations as the previous
+        # version of ctapipe:
+        pedestal_stddev = 1.1*np.ones(len(image))
+
+        muonintensityoutput = \
+            intensity_fitter(1,
+                             muonringparam.ring_center_x,
+                             muonringparam.ring_center_y,
+                             muonringparam.ring_radius,
+                             image*dist_mask,
```

Comment:
This is not going to work well... in this way there will be many pixels with 0 charge but they will be used in the muon ring likelihood fit (sure, they should be pixels with ~0 predicted light, but we want to ignore them completely). It seems this new ctapipe implementation does not allow to set a mask, or pass an image with only some pixels, since it always uses the full camera geometry, as far as I understand. Something to fix on the ctapipe side, do you agree, @MaxNoe ?

Reply:
but what about performing a cleaning and passing the `clean_mask`?

Reply:
> but what about performing a cleaning and passing the `clean_mask`?

You mean masking the image array (it is an ndarray) before passing it? Would not work, because you need to pass the mask as well so that it knows which pixels' coordinates to use. Or am I missing anything?

Reply:
maybe I'm missing the problem here: what you want is that the pixels set to 0 in the `image * dist_mask` are not used by the likelihood whatsover?